### ISSUE TITLE
Editing captions in the Gallery Block creates conflict

### ISSUE THREAD

**Issue by marianearchimbaud** [Automattic/wp-calypso#46813]
Date from the user's report: October 26th

After editing some captions from the images in the gallery block, images are not visible anymore

#### Steps to reproduce
1. Add a gallery block.
2. Edit the image captions from the Gallery block on the editor.
3. Images are replaced by a white/gray square. Screenshot: https://prnt.sc/v79cse
4. If images are replaced, a link is displayed. Screenshot: https://prnt.sc/v78lf1

5. I created a new test post on the user's site, and the result was the same. Images disappear after adding/editing the image caption on the editor.
6. Images and captions are displayed correctly when editing captions on the media library. Screenshot: https://prnt.sc/v79mn9

7. Suggested the user editing the image caption in the media library before placing them in a gallery block, and replace the gallery blocks for new ones. It worked, but there are other posts with the same issue too:

http://bitchinchickens.com/2020/10/01/case-study-burn-survivors/ Screenshot: https://prnt.sc/v7ytvp

#### Context / Source
25176210-hc
3438940-zd

#### Screenshot:
https://prnt.sc/v78lf1
https://prnt.sc/v79cse
https://prnt.sc/v7ytvp

**Comment by lizkarkoski** [comment]
Looks like this is the same issue appearing on https://jgeilsband.wordpress.com/j-geils-band-atlantic-tapes-1970-1982/
Reported in the WordPress.com forums: https://wordpress.com/forums/topic/gallery-block-problem/

**Comment by ockham** [comment]
I can reproduce on WP.com. The crucial part is to have a link in the image caption :+1: 
To clarify, the issue is visible on the frontend (but not on the editor).

Going to test on vanilla Gutenberg now.

**Comment by ockham** [comment]
Doesn't seem present in vanilla Gutenberg, so I'm suspecting some of WP.com's sanitization (`wp_kses` filters?), apparently in combination with something that changed in Gutenberg 9.2 over previous versions.

On WP.com, the `<img />` tag is wrapped in an `<a />`, and is escaped:

```html
<figure class="wp-block-gallery columns-3 is-cropped">
    <ul data-carousel-extra='{"blog_id":89417913,"permalink":"https:\/\/example.com\/2020\/10\/28\/gallery\/"}'
        class="blocks-gallery-grid">
        <li class="blocks-gallery-item">
            <figure><a
                    href="https://example.files.wordpress.com/2019/02/cat-pet-animal-domestic-104827-1.jpeg">&lt;img
                    src=&quot;https://example.files.wordpress.com/2019/02/cat-pet-animal-domestic-104827-1.jpeg?w=580&quot;
                    alt=&quot;&lt;a href=&quot;https://example.com/&quot;&gt;Caption</a></a>
                <figcaption class="blocks-gallery-item__caption"><a
                        href="https://example.com/">Caption</a></figcaption>
            </figure>

```

**Comment by synora** [comment]
Another report of this 3445774-zen. The user will try the workaround mentioned in step 7 above.

**Comment by ockham** [comment]
Potentially relevant GB PR: https://github.com/WordPress/gutenberg/pull/26082

**Comment by supernovia** [comment]
Noting a couple of cases: https://wordpress.com/forums/topic/galleries-not-showing-up-right/
https://wordpress.com/forums/topic/unreadable-appearance-of-gallery-page-after-editing/

**Comment by macmanx2** [comment]
Another case: https://wordpress.com/forums/topic/cant-make-changes-to-website-and-drop-down-menus-are-not-working/

**Comment by macmanx2** [comment]
Another: https://wordpress.com/forums/topic/galleries-not-showing-up-right/

**Comment by designsimply** [comment]
Another report:

> Good Afternoon... on my page Shop Amazon theres been an issue with viewing the image and the link...

<sup>(internal reference: 3446502-zen)</sup>

![Screen Shot 2020-10-28 at Oct 28 6 30 14 PM](https://user-images.githubusercontent.com/1119271/97511014-a536b680-194b-11eb-9c34-2b1f7982a514.png)
<sup>Tested with Chrome 85.0.4183.121 on macOS 10.15.6.</sup>

I tested and confirmed the bug, but I had to try editing captions a few different times and the problem seemed to happen after editing a pre-existing gallery, adding a new image, and then making the entire caption into a link.

Post: https://made4testing0318.blog/2020/10/28/gallery/

Screenshot:

<img src="https://user-images.githubusercontent.com/1119271/97511617-1dea4280-194d-11eb-99f9-d47b73ca4db2.png" width="400" alt="Screenshot_2020-10-28 Gallery">
<sup>Tested with Firefox 81.0.1 on macOS 10.15.6.</sup>

**Comment by ockham** [comment]
Filed a request to fix the underlying GB issue: https://github.com/WordPress/gutenberg/issues/26549

We might be able to tackle this at JP level, to get an expedited workaround.

**Comment by jjchrisdiehl** [comment]
Seems to be an issue for captions that are added via the editor that do not exist in the media library. Example: https://personaltestsite123.wordpress.com/2020/10/29/2031/

Middle caption was added in the media library, hyperlink works fine. The third image I added the caption in the editor and it breaks the image.

**Comment by ockham** [comment]
> Seems to be an issue for captions that are added via the editor that do not exist in the media library. Example: p9RHoS-wL-p2
> 
> Middle caption was added in the media library, hyperlink works fine. The third image I added the caption in the editor and it breaks the image.

Yeah, captions set from the media library are really `alt` attributes, which by definition are text-only. Captions set in the gallery editor are rich text and can include HTML (e.g. links). Since https://github.com/WordPress/gutenberg/pull/26082, GB has used the gallery-set caption as a fallback for the `alt` if it's not set through the media library. Unfortunately, it doesn't strip the HTML, which causes the issue.

We're working on a fix here: https://github.com/WordPress/gutenberg/pull/26574

**Comment by dcoleonline** [comment]
Another in 23487450-hc. Will follow up once this is patched.

**Comment by as-esu** [comment]
Another related ticket zen-3450533

**Comment by ash1eygrace** [comment]
Another related: hc-25305530

**Comment by gwensmithx** [comment]
Another issue here: #3447618-zd (simple site)

Can replicate on my test site.

Screenshots of Editor and front-end:

<img width="862" alt="Screen Shot 2020-10-31 at 1 28 35 PM" src="https://user-images.githubusercontent.com/5779512/97788231-17daa880-1b7d-11eb-8db1-2e9a670aa3f5.png">
<img width="650" alt="Screen Shot 2020-10-31 at 1 28 17 PM" src="https://user-images.githubusercontent.com/5779512/97788234-1a3d0280-1b7d-11eb-815e-acd9721b71ca.png">

**Comment by lizkarkoski** [comment]
Report from the WordPress.com forums: https://wordpress.com/forums/topic/lost-image-2/

Screen shot: https://d.pr/i/QWnhrH

**Comment by lizkarkoski** [comment]
Report from the WordPress.com forums: https://github.com/Automattic/wp-calypso/issues/46813 There are linked screen shots in that thread.

**Comment by poojashetty18** [comment]
Another issue here: #24990893-hc (simple site)

Screenshots attached of the Editor and frontend page:

Editor page:
<img width="1159" alt="Screenshot on 2020-11-02 at 11-14-50" src="https://user-images.githubusercontent.com/44837596/97840437-6a42d300-1cfd-11eb-82c4-9a55cdca7494.png">

Frontend page:

<img width="1090" alt="Screenshot on 2020-11-02 at 11-15-19" src="https://user-images.githubusercontent.com/44837596/97840451-729b0e00-1cfd-11eb-8414-1039629bfe49.png">

**Comment by khristiansnyder** [comment]
Another case in #23248587-hc.
Showing in both editor and front end.

**Comment by gwensmithx** [comment]
Maybe this should have a high priority label? I added it but it didn't seem like there were enough cases yet.

Another one here: #23545276-hc 

Showing in Editor and also front-end. Simple site on Stow theme.

**Comment by ockham** [comment]
Fixed by https://github.com/WordPress/gutenberg/pull/26676, which will be part of Gutenberg 9.3, and hopefully land on WP.com soon.

This will probably not un-break gallery blocks that had been broken by this issue. Instead, users will probably have to revert to the last working revision of the affected post.

**Comment by xpurichan** [comment]
Reported in 15739493-hc.

**Comment by susanjanec** [comment]
Another report in 3455554-zd -- showing in editor and front end.

**Comment by khristiansnyder** [comment]
Another report in 3452471-zd

**Comment by abcrsol** [comment]
Another case reported here 3458054-zd

**Comment by abcrsol** [comment]
Reported also in 3462496-zd

![image](https://user-images.githubusercontent.com/73123045/98258205-e0a33780-1f80-11eb-8782-1d945c5d15f6.png)

**Comment by allanftd** [comment]
Hi, I have some more inputs from the user who reported the original issue on this thread:
- I opened this published post to edit a category and when I updated all the photos look like this: https://bitchinchickens.com/2020/09/21/chicken-mating-injuries/
- I opened this published post and when I updated one photo disappeared: https://bitchinchickens.com/2020/10/15/how-to-preserve-eggs/
- If I open and close a published post something happens with the photos regardless of whether I made an update or not, whether it was to add a photo or to make edits to text -- just opening the post is enough to trigger the problem with the photos.

**Update: Looks like the bug fix has already taken effect on her website.**

Just now, I tried reproducing the bug using a test post on the user's site but I can no longer replicate the issue. Here's the test page https://bitchinchickens.com/2020/10/27/test-jade-artist-for-animals/. The last Image gallery with three pictures captioned "nov6" is the most recent one I used. 

Looks like it is working perfectly on my tests but still awaiting feedback from user to confirm.

**Comment by susanjanec** [comment]
Another report in 3466824-zen

**Comment by wprhunt** [comment]
Got another. They're using the TextBook theme:
25460678-hc

**Comment by gwensmithx** [comment]
Another here: #21950117-hc and #3470026-zd

Is there any workaround that anyone can find in the meantime without us having to completely recreate the page? I thought maybe removing the captions in the HTML might work but that doesn't seem to help.

**Comment by scosgro** [comment]
Another report in #3471922-zen

**Comment by poojashetty18** [comment]
Another report in 3471828-zen

**Comment by ockham** [comment]
> Update from original user who reported this issue: she tried Tiled Gallery and there were no issues uploading and publishing the images. However, captions are not supported so she would still prefer using Image Gallery block.
> 
> > Fixed by [WordPress/gutenberg#26676](https://github.com/WordPress/gutenberg/pull/26676), which will be part of Gutenberg 9.3, and hopefully land on WP.com soon.
> > This will probably not un-break gallery blocks that had been broken by this issue. Instead, users will probably have to revert to the last working revision of the affected post.
> 
> Hi @ockham, should the update apply by now on WordPress.com? Please advise.
> Thanks for your help on this!

Hey @allanftd, we're still working on getting the 9.3 upgrade ready for WordPress.com. I'll update this issue once we've landed it.

**Comment by jonnyruzek** [comment]
3478625-zen

<img width="1533" alt="Screen Shot on 2020-11-10 at 08-33-09" src="https://user-images.githubusercontent.com/39252765/98696114-a515be00-2330-11eb-9786-2b4a2761c2f9.png">

**Comment by allanftd** [comment]
Thank you, @ockham! :)

**Comment by fresatomica** [comment]
Another report in https://wordpress.com/forums/topic/gallery-glitch-2/.

**Comment by edequalsawesome** [comment]
Also reported in 25568823-hc, follow-up for the user set up here: 3481646-zen

**Comment by JessBoctor** [comment]
Another report of this issue in 14297746-HC. Set up follow up here: 3482074-Zen

**Comment by jerrysarcastic** [comment]
Also reported in https://wordpress.com/forums/topic/image-links-not-working-2

**Comment by wpniall** [comment]
Update from 3438940-zd:

User said they scheduled a post and the photos disappeared after hitting the preview button. Only the captions boxes were left. 
Opening the editor showed that the photos were still there. So the photos appear in the editor but not when previewed.

When they tried again, the images completely disappeared from the editor as happened in the original issue.

**Comment by adminhappymondays** [comment]
One more report here (in Spanish, sorry): https://wordpress.com/es/forums/topic/error-en-bloque-galeria/

After many tests, in my case, the problem appears when I convert a simple text caption of a picture into a link through the visual editor. It looks like the HTML conversion of the changes I made into the gallery has changed. Some of the content of the <img> tags disappeared (I don't know why) and if you try to see the code, includes so much mistakes.

here an example: https://happymondays.es/actrices-2/
here another one with and without a link: https://happymondays.es/pruebas-galeria-nueva/

Hope the developer can fix the problem SOON.
Thanks

**Comment by poojashetty18** [comment]
Another report in 3489909-zen

**Comment by wpniall** [comment]
Another report in 25666333-hc. Follow-up at 3493156-zen.

**Comment by ockham** [comment]
This should no longer occur per D52264-code (p9o2xV-19f-p2 for Atomic). Existing broken galleries unfortunately have to be fixed manually (e.g. by reverting to a previous revision of the post or page).

**Comment by ockham** [comment]
Reopening, since we had to revert the 9.3.0 upgrade :confused:

**Comment by ajaykj** [comment]
Another report in 25687636-hc. 

<img width="870" alt="Markup 2020-11-17 at 10 11 43" src="https://user-images.githubusercontent.com/35197279/99350044-73b66880-28c3-11eb-8c4e-e9557a6c39f1.png">

Follow-up at 3495510-zen.

**Comment by jonnyruzek** [comment]
25728275-hc / 3500403-zen

**Comment by lakellie** [comment]
Closing this per the discussion in p1605780791157700-slack-wpcom-happy-announce. This was resolved with the re-release of Gutenberg 9.3 on simple sites. 

This release will not resolve the issue on existing galleries. To fix those, users will need to either:
1. Recreate the gallery
2. Restore a revision of the page from when the gallery was working
